### ISSUE TITLE
App versions

### ISSUE THREAD

**Issue by throttlemeister** [IceWhaleTech/CasaOS#1180]
**Is your feature request related to a problem? Please describe.**
When you install an app from the app store, it pulls a pre-specified version that you cannot change from Docker Hub. This is a problem, as these versions may not operate as expected or flatout not work properly.

Example: Vaultwarden, the version installed is old and does not operate with the current browser extensions. The only way to fix this is to run the latest version, so 1) install manually, or 2) (as in my case) just delete from CasaOS start on my Synology again.

**Describe the solution you'd like**
Being able to specify and/or change the version pulled from Docker, either before installing or after installing.

**Describe alternatives you've considered**
Alternative is not using CasaOS and just use whatever Docker host one wants to use and do it manually or through tools like Synology Docker or Portainer.

This is, in my humble opinion, hurting CasaOS adoption. I for one, after these experiences and spending hours troubleshooting why something wasn't working, am more inclined to just run containers on my Synology host and not on CasaOS.

**Comment by WisdomSky** [comment]
I've also made a couple third-party CasaOS appstore that constantly checks for newer versions and update the listing regularly.

* For [LinuxServer.io](https://fleet.linuxserver.io/) Docker Images:
https://github.com/WisdomSky/CasaOS-LinuxServer-AppStore

* Forked version of the Main CasaOS AppStore that also tracks for new versions and updates the list: 
https://github.com/WisdomSky/CasaOS-AppStore-Edge
> Note: Unlike the Main CasaOS AppStore which only updates versions to the latest MINOR versions, the forked one will do MAJOR version updates, so if an app has released a major version jump like 3.X.X into 4.X.X, it will also update to that version which could mean there are breaking changes and your current installed app may end up not working after updating. So use at your own risk or be sure to make necessary back-up to your app's config and files before updating.

**Comment by throttlemeister** [comment]
Yes sir, I was still on the old version despite removing / reinstall and update check.

But why can't we just either use latest or manual change the version we want to use if we want to deviate from default for whatever reason.

This seems unnecessarily restrictive to me.

**Comment by throttlemeister** [comment]
I am on the latest released aka v0.43 btw. You can also not see which version is going to be installed, until you are actually installing. Now, if you are always on the latest version, that may be ok. But if you are not, it would be nice to know so you can verify you are not missing features you need or expect because the respective website for the app tell you it is.

**Comment by ETWang1991** [comment]
![应用商店详情页自定义安装](https://github.com/IceWhaleTech/CasaOS/assets/109786132/15a6e7b3-e623-4f5c-8e88-398ef44496fa)
Maybe this is what you guys want

The CasaOS team has been busy with the launch of the new hardware products and will try to update all the apps in the app store when  finished.

**Comment by throttlemeister** [comment]
If you can change all field before install, that'd be a great option

**Comment by LinkLeong** [comment]
> If you can change all field before install, that'd be a great option

Do you mean we update automatically? Or do you choose the version number?

**Comment by raphamotta** [comment]
> ![应用商店详情页自定义安装](https://user-images.githubusercontent.com/109786132/245732528-15a6e7b3-e623-4f5c-8e88-398ef44496fa.png) Maybe this is what you guys want
> 
> The CasaOS team has been busy with the launch of the new hardware products and will try to update all the apps in the app store when finished.

Yes, yes! A custom install will be perfect. I like to use watchtower to update my containers, so I'm always installed manually to use the latest version, never from app store

**Comment by NiklasOlofsson** [comment]
> If you can change all field before install, that'd be a great option

Yes or even after install. I don't understand the version pinning but I'm very new to casaos.

**Comment by Karoolus** [comment]
I'm inspecting the field in developer console and removing the readonly flag. That way I can change the field value and it'll update to the version I want it to be on. This is a ridiculous workaround and I don't understand why we don't have full control of our own apps?

**Comment by mr-manuel** [comment]
@nesalkiN see https://github.com/IceWhaleTech/CasaOS-AppStore/issues/167

But I'm also the opinion that for some apps the `:latest` tag would neverthless work fine. Like containers that does not need a config and are auto updating their internals.

**Comment by lothriell** [comment]
having same issue with unify controller version..
